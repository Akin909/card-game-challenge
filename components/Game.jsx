import React, { Component } from 'react';
import uuid from 'uuid';
import { Deck, dealCards, replay, hand } from './GameLogic';

class Game extends Component {
  state = {
    value: '',
    cards: Deck,
  };

  handleClick = event => {
    //This returns an array of objects which represents the deck at each
    //point in the shuffle
    const newDeck = Array.from({ length: Number(this.state.value) }, () =>
      dealCards(this.state.cards)
    );
    //Set state to the deck at the end of the shuffle
    this.setState({ cards: newDeck.slice(-1) });
  };

  handleChange = event => this.setState({ value: event.target.value });

  render() {
    console.log('hand', hand);
    return (
      <div>
        <button onClick={this.handleClick}>Deal a Hand</button>
        <input
          type="text"
          placeholder="Number of hands"
          value={this.state.value}
          onChange={this.handleChange}
        />
        <div>
          {Object.keys(this.state.cards).map(key => (
            <div key={uuid()}>
              <p>{key}</p>
              <pre>{JSON.stringify(this.state.cards[key], undefined, 2)}</pre>
            </div>
          ))}
          <ul>
            {replay.map(step => <li key={uuid()}>{step}</li>)}
          </ul>
        </div>
      </div>
    );
  }
}

export default Game;
