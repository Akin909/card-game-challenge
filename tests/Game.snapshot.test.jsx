import React from 'react';
import renderer from 'react-test-renderer';
import Game from './../components/Game.jsx';

describe('Game component renders correctly', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(<Game />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
