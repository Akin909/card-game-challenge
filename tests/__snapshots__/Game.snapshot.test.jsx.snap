// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game component renders correctly renders correctly 1`] = `
<div>
  <ul
    className="kscWej"
  >
    <h2>
      Rules
    </h2>
    <li>
      Each number card is worth one point.
    </li>
    <li>
      Each Face card has a special value.
      <ul>
        <li>
          King: 13
        </li>
        <li>
          Queen: 12
        </li>
        <li>
          Jack: 11
        </li>
      </ul>
    </li>
    <li>
      A Pair i.e. two cards with the same value are worth 10 extra points
    </li>
    <li>
      Three of a kind - 3 cards with the same value are worth 20 extra points
    </li>
    <li>
      A Straight - all cards with the same value are worth 40 extra points
    </li>
  </ul>
  <h2
    className="jUyNoj"
  />
  <section
    className="ebfgaU"
  >
    <input
      className="czvjFH"
      name="input"
      onChange={[Function]}
      placeholder="Number of hands"
      type="text"
      value={undefined}
    />
    <select
      className="hMHZdq"
      id="players"
      name="select"
      onChange={[Function]}
      size={5}
      value="2"
    >
      <option
        value={2}
      >
        2 player(s)
      </option>
      <option
        value={3}
      >
        3 player(s)
      </option>
      <option
        value={4}
      >
        4 player(s)
      </option>
      <option
        value={5}
      >
        5 player(s)
      </option>
      <option
        value={6}
      >
        6 player(s)
      </option>
      <option
        value={7}
      >
        7 player(s)
      </option>
      <option
        value={8}
      >
        8 player(s)
      </option>
      <option
        value={9}
      >
        9 player(s)
      </option>
      <option
        value={10}
      >
        10 player(s)
      </option>
      <option
        value={11}
      >
        11 player(s)
      </option>
      <option
        value={12}
      >
        12 player(s)
      </option>
      <option
        value={13}
      >
        13 player(s)
      </option>
      <option
        value={14}
      >
        14 player(s)
      </option>
      <option
        value={15}
      >
        15 player(s)
      </option>
      <option
        value={16}
      >
        16 player(s)
      </option>
      <option
        value={17}
      >
        17 player(s)
      </option>
      <option
        value={18}
      >
        18 player(s)
      </option>
      <option
        value={19}
      >
        19 player(s)
      </option>
      <option
        value={20}
      >
        20 player(s)
      </option>
      <option
        value={21}
      >
        21 player(s)
      </option>
      <option
        value={22}
      >
        22 player(s)
      </option>
      <option
        value={23}
      >
        23 player(s)
      </option>
      <option
        value={24}
      >
        24 player(s)
      </option>
      <option
        value={25}
      >
        25 player(s)
      </option>
      <option
        value={26}
      >
        26 player(s)
      </option>
    </select>
    <button
      onClick={[Function]}
    >
      Deal a Hand
    </button>
  </section>
  <div
    className="jvjBkq"
  >
    <span
      className="hdKDHS"
    />
  </div>
  <div>
    <ul
      className="eVryjL"
    />
  </div>
</div>
`;
